### PR TITLE
Author show

### PR DESCRIPTION
**Pull request by treble37**
Implemented the first part of this feature (https://github.com/ChillinAstroTurtles/astro-reads/issues/13) completely.    Still need to show the sample covers.

### CODE REVIEW

File: app/models/book_author.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class BookAuthor < ActiveRecord::Base
```
Comment (dstrunk) [inline app/models/book_author.rb]: I don't think we need a `through` table here... unless you're thinking a book _should_ have many authors? For me, it makes more sense to have `book belongs_to :author`, and `author has_many :books`. Having a through table seems like premature normalization, and I don't see us needing to reference the BookAuthor relationship _directly_.

If we do think a join table is necessary, can we consider a different name? Something like "Credit" which is easier to hold in our heads.

Comment (treble37) [inline app/models/book_author.rb]: Thanks @dstrunk, I probably should have run this by you guys first.  I was thinking about the case of co-authors: https://www.goodreads.com/book/show/67035.Relic?from_search=true&search_version=service  ( Douglas Preston (Goodreads Author), Lincoln Child) - what do you think?

Comment (dstrunk) [inline app/models/book_author.rb]: That's fine! At first glance it just looked like premature optimization.

What do you think about renaming the model to "Credit" to make more sense from a vocabulary standpoint?

File: config/routes.rb [hunk 2]
```diff
@@ -1,3 +1,5 @@
 Rails.application.routes.draw do
   resources :books, only: [:index]
+  resources :authors, only: [:show]
+  get "/authors/:id/books", to: "authors#books"
```
Comment (dstrunk) [inline config/routes.rb]: This non-RESTful route might be uneccessary. For an MVP, and following the specs on Upcase, I think it's fine to just render up to 5 books on the Author `show` page (perhaps in a book partial, after initially implemented without a partial).

Comment (treble37) [inline config/routes.rb]: Thanks @dstrunk, good catch.  I'll refactor it out and make a commit showing the edit.

File: app/models/book_author.rb (hunk 1 above)
Comment (treble37) [inline app/models/book_author.rb]: Naming is hard...credit actually made me think of it in a financial sense before I thought about it further....but I'm fine with it if the group decides that's a name for it.  Maybe we should talk about it on the call tonight.

**Comment by treble37** [comment]
@dstrunk , @mrlee-io  - please take a look at item 2 below especially:

1) Added cover_image_url to Book
2)  Hound CI is complaining about preferring has_many :through - should we disregard for now and merge this in?

**Comment by treble37** [comment]
@dstrunk, thanks I fixed the array, the thoughtbot guide says to avoid collection/member....with that, are you ok to merge then?

**Comment by dstrunk** [comment]
:+1:

**Pull request merged**